eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/TagManager.java

```diff
@@ -0,0 +1,870 @@
+      new BasicProperty<>(REALTIME_OVERRIDE_PROPERTY_NAME, Boolean.class);
+
+  /**
+   * The name of the property used to store the raw Ewon tag name, for an Ewon Flexy tag via M2Web,
+   * inside an Ignition tag.
+   *
+   * @since 1.0.0
+   */
+  private static final String RAW_EWON_TAG_NAME_PROPERTY_NAME = "RawEwonTagName";
+
+  /**
+   * The tag property used to store the raw Ewon tag name, for an Ewon Flexy tag via M2Web, inside
+   * an Ignition tag.
+   *
+   * @since 1.0.0
+   */
+  private static final BasicProperty<String> RAW_EWON_TAG_NAME_PROPERTY =
+      new BasicProperty<>(RAW_EWON_TAG_NAME_PROPERTY_NAME, String.class);
+
+  /**
+   * The Ewon connector gateway context.
+   *
+   * @since 1.0.0
+   */
+  private static GatewayContext gatewayContext;
+
+  /**
+   * The Ewon connector settings.
+   *
+   * @since 1.0.0
+   */
+  private static EwonConnectorSettings connectorSettings;
+
+  /**
+   * Tag provider for configured tags
+   *
+   * @since 1.0.0
+   */
+  private static ManagedTagProvider tagProvider;
+
+  /**
+   * Initializes the tag manager using the provided {@link GatewayContext} and {@link
+   * EwonConnectorSettings}.
+   *
+   * @param gatewayContext the gateway context
+   * @param connectorSettings the Ewon connector settings.
+   * @since 1.0.0
+   */
+  public static void initialize(
+      GatewayContext gatewayContext, EwonConnectorSettings connectorSettings) {
+    // Store gateway context and connector settings
+    TagManager.gatewayContext = gatewayContext;
+    TagManager.connectorSettings = connectorSettings;
+
+    // Create realtime tag provider
+    tagProvider =
+        gatewayContext
+            .getTagManager()
+            .getOrCreateManagedProvider(
+                new ProviderConfiguration(GATEWAY_TAG_PROVIDER_NAME)
+                    .setAllowTagCustomization(true)
+                    .setPersistTags(true)
+                    .setPersistValues(true)
+                    .setAllowTagDeletion(true));
+
+    // Initialize status tags
+    StatusTagManager.initializeStatusTags(tagProvider);
+
+    // Configure tag provider deletion handler to cleanup tags on deletion
+    tagProvider.setDeletionHandler(
+        tagPath -> {
+          // Get Ewon Flexy name from tag path
+          final int tagPathComponentIndexEwonName = 0;
+          String ewonName = tagPath.getPathComponent(tagPathComponentIndexEwonName);
+
+          // Get Ewon Flexy tag name from tag path
+          final int tagPathComponentIndexTagName = 1;
+          String tagName = tagPath.getPathComponent(tagPathComponentIndexTagName);
+
+          // Remove from initialized tags
+          if (INITIALIZED_EWON_TAGS.containsKey(ewonName)) {
+            INITIALIZED_EWON_TAGS.get(ewonName).remove(tagName);
+            if (INITIALIZED_EWON_TAGS.get(ewonName).isEmpty()) {
+              INITIALIZED_EWON_TAGS.remove(ewonName);
+              FORCED_M2WEB_EWONS.remove(ewonName);
```

Comment:
ewonName is string, 
not sure if you can remove string from Collection<M2WebEwon>. I see a  Collection.remove(Object), but not Collection.remove(string)

Reply:
Will correct